File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer,Integer> map = new HashMap<>();

        int count = 0;
        int currPrefixSum = 0;

        map.put(0,1);

        for(int num : nums) {
            currPrefixSum += num;

            int target = currPrefixSum - k;

            if(map.containsKey(target)) {
                count += map.get(target);
            }

            map.put(currPrefixSum, map.getOrDefault(currPrefixSum, 0) + 1);
        }

        return count;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        have_contained = {}

        for i, num in enumerate(nums):
            diff = target - num
            if diff in have_contained:
                return [have_contained[diff], i]
            have_contained[num] = i     